etcdserver/v3demo_server.go

```diff
@@ -301,8 +301,19 @@ func applyCompare(txnID int64, kv dstorage.KV, c *pb.Compare) (int64, bool) {
 		}
 		return rev, false
 	}
-
-	ckv := ckvs[0]
+	var ckv storagepb.KeyValue
+	if len(ckvs) != 0 {
+		ckv = ckvs[0]
+	} else {
+		// Use the zero value of ckv normally. However...
+		if c.Target == pb.Compare_VALUE {
```

Comment:
I think we do not have to do this. If we want to compare the existence, we can always compare the version, right?